tests/Resources/core.runtime.addontest.js

```diff
@@ -0,0 +1,114 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2011-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti */
+/* eslint no-unused-expressions: "off" */
+
+'use strict';
+
+var should = require('./utilities/assertions');
+
+describe('Core', () => {
+	describe('Runtime', () => {
+		describe('hasProperty', () => {
+			describe('Top-Module', () => {
+				describe('Submodules', () => {
+					it('should check for sub-module', () => {
+						Ti.should.have.property('UI');
+						// @fixme this check throws an error on android
+						if (Ti.Platform.osname !== 'android') {
+							Ti.should.not.have.property('Foo');
+						}
+					});
+				});
+
+				describe('Custom properties', () => {
+					it.ios('should check for custom properties', () => {
```

Comment:
why limit this to iOS only?

Reply:
Android throws an error when doing `.not.have.property` checks directly on modules. I'll mark it as `it.androidBroken` together with the other one above.